api/common/id.go

```diff
@@ -18,3 +18,40 @@
 package common
 
 type ChunkID uint64
+
+type ChunkIDs []ChunkID
+
+// HashIntersect returns an intersection of two ChunkID arrays
+// without any assumptions on the order. It uses a HashMap to mark
+// the existence of a item.
+func (c ChunkIDs) HashIntersect(other ChunkIDs) ChunkIDs {
+	if c == nil && other == nil {
```

Comment:
The receiver `c` should not be nil from what I know. Can we remove this precondition?

Reply:
Fixed